### PR TITLE
`gpaa-single-line-as-input.php`: Added full address capability to the Single Line as Input functionality.

### PR DESCRIPTION
**Pull request by saifsultanc**
## Context

⛑️ Ticket(s): https://secure.helpscout.net/conversation/2478783845/59879?folderId=7098280

## Summary

Use [full street address](https://github.com/gravitywiz/snippet-library/blob/master/gp-address-autocomplete/gpaa-use-full-street-address.js) snippet cannot be directly used with the current Single Line as Input snippet. This update adds an extra argument to the Single Line as Input `use_full_address` and the internal logic with `gpaa_values` processes either functionality accordingly.

Quick Loom:
https://www.loom.com/share/b0633a46e44547189ca51954d524d5c4

### CODE REVIEW

File: gp-address-autocomplete/gpaa-single-line-as-input.php [hunk 1]
```diff
@@ -55,8 +55,13 @@ public function output_script() {
 				window.GPAASingleLineInput = function( args ) {
 
 					gform.addFilter('gpaa_values', function (values, place) {
-						values.autocomplete = place.formatted_address;
-
+						if ( args.useFullAddress ) {
+							var fullAddress     = instance.inputs.autocomplete.value;
```
Comment (claygriffiths) [inline gp-address-autocomplete/gpaa-single-line-as-input.php:59]:
```suggestion
							// Logic borrowed from https://github.com/gravitywiz/snippet-library/pull/730
							var fullAddress     = instance.inputs.autocomplete.value;
```

File: gp-address-autocomplete/gpaa-single-line-as-input.php [hunk 2]
```diff
@@ -107,5 +113,6 @@ public function is_applicable_form( $form ) {
 new GPAA_Single_Line_Input( array(
 	'form_id'              => 123,     // The ID of your form.
 	'address_field_id'     => 4,       // The ID of the Address field.
-	'single_line_field_id' => 5,        // The ID of the Single Line Text field.
+	'single_line_field_id' => 5,       // The ID of the Single Line Text field.
+	'use_full_address'     => true,    // Whether to use full street address or not
```
Comment (claygriffiths) [inline gp-address-autocomplete/gpaa-single-line-as-input.php:117]:
```suggestion
	'single_line_field_id' => 5,       // The ID of the Single Line Text field.
	// 'use_full_address'     => true,    // Uncomment to use the full street address if you don't want an abbreviated street address.
```

(it's hard to get whitespace right in this suggestion 😆)

**Review by claygriffiths — COMMENTED** [review]
Looks fine to me! I have a couple of suggestions to help clarify things for users and future us.

**Comment by saifsultanc** [comment]
PR updated: https://github.com/gravitywiz/snippet-library/pull/818

**Pull request closed without merging**